specs/lib/writeResourcesSpec.js

```diff
@@ -9,14 +10,26 @@ const Cesium = require('cesium');
 const CesiumMath = Cesium.Math;
 
 const gltfPath = 'specs/data/2.0/box-techniques-embedded/box-techniques-embedded.gltf';
+const gltfWebpPath = 'specs/data/2.0/extensions/EXT_image_webp/box-textured-separate/box-textured-with-fallback.gltf';
 let gltf;
+let gltfWebp;
 
 describe('writeResources', function() {
     beforeEach(function(done) {
         const gltfLoaded = fsExtra.readJsonSync(gltfPath);
+        const gltfWebpLoaded = fsExtra.readJsonSync(gltfWebpPath);
         return readResources(gltfLoaded)
             .then(function() {
                 gltf = gltfLoaded;
+            })
+            .then(function() {
+                const options = {
+                    resourceDirectory: path.dirname(gltfWebpPath)
+                };
+                return readResources(gltfWebpLoaded, options);
+            })
+            .then(function() {
+                gltfWebp = gltfWebpLoaded;
```

Comment:
`Promise.all` will make this cleaner.

```js
gltf = fsExtra.readJsonSync(gltfPath);
gltfWebp = fsExtra.readJsonSync(gltfWebpPath);

Promise.all([
    readResources(gltfLoaded),
    readResources(gltfWebpLoaded)
])
    .then(function() {
        done();
    });
```

`resourceDirectory` is not needed if the file has separate resources.